### PR TITLE
Implemented login and callback functionalities

### PR DESCRIPTION
**Pull request by OttoAhlfors**

### CODE REVIEW

File: client/app/page.tsx [hunk 1]
```diff
@@ -1,26 +1,68 @@
...
+  )
+}
+
+function HomeContent() {
+  const [token, setToken] = useState('')
+
+  const queryParams = useSearchParams()
+  const code = queryParams.get('code')
+  const state = queryParams.get('state')
+  const client_redirect_uri = 'http://localhost:80'
```
Comment (tuomasmustakallio) [inline client/app/page.tsx:21]: Should this port be 8080? This variable could also be out of this function

**Review by tuomasmustakallio — CHANGES_REQUESTED** [review]

File: client/app/page.tsx (hunk 1 above)
Comment (EddieTheCubeHead) [reply client/app/page.tsx:21]: This is the client URI, not server one, so 80 is fine

Comment (EddieTheCubeHead) [reply client/app/page.tsx:21]: Realized there's two questions so unresolving for the second part. 80 is still correct though

**Review by tuomasmustakallio — APPROVED** [review]

File: client/app/page.tsx (hunk 1 above)
Comment (tuomasmustakallio) [reply client/app/page.tsx:21]: Ok thanks

**Pull request merged**